src/main/java/io/logz/logback/LogzioSender.java

```diff
@@ -125,6 +127,26 @@ public LogzioSender(String logzioToken, String logzioType, int drainTimeout, int
         debug("Created new LogzioSender class");
     }
 
+    public static LogzioSender getOrCreateSenderByType(String logzioToken, String logzioType, int drainTimeout, int fsPercentThreshold, String bufferDir,
+                                                       String logzioUrl, int socketTimeout, int connectTimeout, boolean debug,
+                                                       LogzioLogbackAppender.StatusReporter reporter, ScheduledExecutorService tasksExecutor,
+                                                       boolean addHostname, String additionalFields, int gcPersistedQueueFilesIntervalSeconds) {
+
+        LogzioSender logzioSenderInstance = logzioSenderInstances.get(logzioType);
```

Comment:
I'll lock it.
The motivation:
I want one buffer per appender. And the only thing that should be different between appenders, is a type. so that's why I create separate buffers per type.
BUT - users not always understand the notion of types at first, and can define multiple appenders on the same type - and this is what I want to protect by the factory.